test/e2e/secondary_network_test.go

```diff
@@ -0,0 +1,286 @@
+		return nil, err
+	}
+	success = true
+	logs.Infof("Checking CoreDNS deployment")
+	if err := testData.checkCoreDNSPods(defaultTimeout); err != nil {
+		return nil, err
+	}
+	return testData, nil
+}
+
+// The function extractPodsInfo extracts the pod and secondary network interface information for the creation of pods from secondary_network_configuration.yaml file
+func (data *TestData) extractPodsInfo() error {
+	var errYamlUnmarshal error
+	secondaryNetworkConfigYaml, _ := os.ReadFile(secondary_network_config_yaml)
+	errYamlUnmarshal = yaml.Unmarshal(secondaryNetworkConfigYaml, &service)
+	if errYamlUnmarshal != nil {
+		return fmt.Errorf("Parsing %s failed.", secondary_network_config_yaml)
+	}
+	interfaceType = service.InterfaceType.InterfaceType
+	totalNumberOfPods = service.CreatePod.Numberofpods
+	logs.Infof("Interface type is : %v", interfaceType)
+	for _, s := range service.CreatePod.Describe {
+		output := describePodInfo{nameOfPods: s[podName].(string), countOfVirtualNetworksPerPod: s[podVNsCount].(int), nameOfVirtualNetworkPerPod: strings.Split(s[podVirtualNetwork].(string), ","), nameOfInterfacePerPod: strings.Split(s[podInterfaceName].(string), ",")}
+		podData = append(podData, output)
+
+	}
+	return nil
+}
+
+// The function formAnnotationStringOfPod forms the annotation string, used in the generation of each pod yaml file.
+func formAnnotationStringOfPod(pod int) string {
+	var annotationString = ""
+	for xPodVN := 0; xPodVN < podData[pod].countOfVirtualNetworksPerPod; xPodVN++ {
+		var podNetworkSpec = "{\"name\": \"" + podData[pod].nameOfVirtualNetworkPerPod[xPodVN] + "\" ,\"interface\": \"" + podData[pod].nameOfInterfacePerPod[xPodVN] + "\" , \"type\": \"" + interfaceType + "\"}"
+		if annotationString == "" {
+			annotationString = "[" + podNetworkSpec
+		} else {
+			annotationString = annotationString + "," + podNetworkSpec
+		}
+	}
+	annotationString = annotationString + "]"
+	return annotationString
+}
+
+// The function creationOfPodOnNode creates the pod for the specific annotations as per the parsed pod information using the NewPodBuilder API
+func (data *TestData) creationOfPodOnNode(ns string, nodeName string) error {
+	labelApp := "testsecpod"
+	for xPod := 0; xPod < totalNumberOfPods; xPod++ {
+		err := data.createPodForSecondaryNetwork(ns, nodeName, xPod, labelApp, resNum)
+		if err != nil {
+			return fmt.Errorf("Error in creating pods.., err: %v", err)
+		}
+	}
+	return nil
+}
+
+// The function getSecondaryInterface shows up the secondary interfaces created for the specific pod and extracts the IP address for the same.
+func (data *TestData) getSecondaryInterface(targetPod int, targetInterface int) (string, error) {
+	respCode, output, _, err := data.RunCommandOnNode(controlPlaneNodeName(), fmt.Sprintf("kubectl exec -it %s -n kube-system -- ip addr show %s | grep \"inet\" | awk '{print $2}' | cut -d/ -f1", podData[targetPod].nameOfPods, podData[targetPod].nameOfInterfacePerPod[targetInterface]))
```

Comment:
if you need to run a command on a Pod, use `RunCommandOnPod`. `RunCommandOnNode` should be avoided as much as possible, as it makes assumptions as to which software is available on each Node.